kimchi/src/verifier_index.rs

```diff
@@ -197,6 +205,16 @@ where
                 array_init(|i| self.srs.commit_evaluations_non_hiding(domain, &c[i], None))
             }),
 
+            foreign_mul_comm: self
+                .cs
+                .foreign_mul_selector_polys
+                .iter()
+                .map(|poly| {
+                    self.srs
+                        .commit_evaluations_non_hiding(domain, &poly.eval8, None)
```

Comment:
I believe this will crash because the domain of `domain` and `poly.eval8` won't match. Even if you use `d8` here it might not be a good idea as this means we'd have to precompute all the lagrange basis in d8 IIRC

Reply:
Certainly doesn't crash in the unit tests.

Reply:
It is the same approach used by other gates, such as ec variable base scalar multiplication, endoscalar mul and chacha.

I'm not really sure I can give the correct answer, but I think it is doing this.
* The selector polynomial is generated by interpolating a polynomial whose degree is related to the number of rows, i.e., domain.d1
* We then evaluate this polynomial again, but this time using the large d8 domain (I think this is for efficiency/performance reasons)

@mrmr1993 Can you please advise?

Reply:
I think @mimoo is right, but the actual bad behaviour is shielded by another error: the verifier never checks that the evaluations match with the commitment (which should happen [like this](https://github.com/o1-labs/proof-systems/blob/60617995a56533af71fb439341046579decec5b6/kimchi/src/verifier.rs#L753)). It would be better to remove the evaluations and just use the commitments, if you are able to.

Reply:
I thought this is creating the commitments
```
            foreign_mul_comm: self
                .cs
                .foreign_mul_selector_polys
                .iter()
                .map(|poly| {
                    self.srs
                        .commit_evaluations_non_hiding(domain, &poly.eval8, None)
                })
                .collect(),
```